test/main-test.js

```diff
@@ -464,5 +464,150 @@ vows.describe('Regexp Tokenizer')
+                  { type: types.CHAR, value: 97 },
+                  { type: types.RANGE, from: 45, to: 45 },
+                ],
+              }],
+            });
+          },
+        },
+        'token.from is a hyphen and the range is preceded by a single character [a\\--\\/]': {
+          topic: ret('[a\\--\\/]'),
+          'Tokenizes correctly': t => {
+            assert.deepStrictEqual(t, {
+              type: types.ROOT, stack: [{
+                type: types.SET, not: false, set: [
+                  { type: types.CHAR, value: 97 },
+                  { type: types.RANGE, from: 45, to: 47 },
+                ],
+              }],
+            });
+          },
+        },
+        'token.from is a hyphen and the range is preceded by a single character [c\\--a]': {
+          topic: ret('[c\\--a]'),
+          'Tokenizes correctly': t => {
+            assert.deepStrictEqual(t, {
+              type: types.ROOT, stack: [{
+                type: types.SET, not: false, set: [
+                  { type: types.CHAR, value: 99 },
+                  { type: types.RANGE, from: 45, to: 97 },
+                ],
+              }],
+            });
+          },
+        },
+        'token.from is a hyphen and the range is preceded by a single character [\\-\\--\\-]': {
+          topic: ret('[\\-\\--\\-]'),
+          'Tokenizes correctly': t => {
+            assert.deepStrictEqual(t, {
+              type: types.ROOT, stack: [{
+                type: types.SET, not: false, set: [
+                  { type: types.CHAR, value: 45 },
+                  { type: types.RANGE, from: 45, to: 45 },
+                ],
+              }],
+            });
+          },
+        },
+        'token.from is a hyphen and the range is preceded by a predefined set [\\w\\--\\-]': {
+          topic: ret('[\\w\\--\\-]'),
+          'Tokenizes correctly': t => {
+            assert.deepStrictEqual(t, {
+              type: types.ROOT, stack: [{
+                type: types.SET, not: false, set: [
+                  {
+                    type: types.SET, not: false, set: [
+                      { type: types.CHAR, value: 95 },
+                      { type: types.RANGE, from: 97, to: 122 },
+                      { type: types.RANGE, from: 65, to: 90 },
+                      { type: types.RANGE, from: 48, to: 57 },
+                    ],
+                  },
+                  { type: types.RANGE, from: 45, to: 45 },
+                ],
+              }],
+            });
+          },
+        },
+        'token.from is a caret and the range is the first item of the set [\\^-9]': {
+          topic: ret('[\\^-9]'),
+          'Tokenizes correctly': t => {
+            assert.deepStrictEqual(t, {
+              type: types.ROOT, stack: [{
+                type: types.SET, not: false, set: [
+                  { type: types.RANGE, from: 94, to: 57 },
+                ],
+              }],
+            });
+          },
+        },
```

Comment:
i get an error when trying to make a regex with this

    SyntaxError: Invalid regular expression: /[\^-9]/: Range out of order in character class

maybe this error check belongs in another PR, but i also don't think this test should suggest this is expected behavior